crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -216,6 +233,63 @@ fn convert_bal(b: sui_json_rpc_types::Balance) -> Balance {
         total_balance: BigInt::from_str(&format!("{}", b.total_balance)).unwrap(),
     }
 }
+pub(crate) async fn convert_to_gas_input(
+    cl: &SuiClient,
+    gas_data: &SuiGasData,
+) -> Result<GasInput> {
+    let payment_obj_ids = gas_data
+        .payment
+        .clone()
+        .iter()
+        .map(|x| x.object_id)
+        .collect::<Vec<_>>();
+
+    let obj_responses = cl
+        .read_api()
+        .multi_get_object_with_options(payment_obj_ids, SuiObjectDataOptions::full_content())
+        .await?;
+
+    let payment_objs = obj_responses
+        .iter()
+        .map(|x| convert_obj(x.data.as_ref().unwrap()))
+        .collect::<Vec<Object>>();
```

Comment:
ditto, on type ascription (also, you don't need to specify the full type, just the collection type, because that is what `collect` is struggling with):

```suggestion
    let payment_objs: Vec<_> = obj_responses
        .iter()
        .map(|x| convert_obj(x.data.as_ref().unwrap()))
        .collect();
```